packages/lambda/deploy.js

```diff
@@ -0,0 +1,223 @@
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+var AWS = require('aws-sdk');
+var hopsConfig = require('hops-config');
+var getAWSConfig = require('./lib/aws-config');
+var createLambdaBundle = require('./lib/create-lambda-bundle');
+var progressWriter = require('./lib/progress-writer');
+var fsUtils = require('./lib/fs-utils');
+
+function formatParameters(params) {
+  return Object.keys(params).map(function (key) {
+    return {
+      ParameterKey: key,
+      ParameterValue: String(params[key])
+    };
+  });
+}
+
+function createBucketIfNotExists (s3, bucketName) {
+  return s3.getBucketLocation({ Bucket: bucketName }).promise()
+    .then(function () { return bucketName; })
+    .catch(function () {
+      return s3.createBucket({ Bucket: bucketName }).promise()
+        .then(function () {
+          return bucketName;
+        });
+    });
+}
+
+function uploadFile (s3, bucketName, file) {
+  var progress = progressWriter('uploading ' + path.basename(file));
+  return fsUtils.hashFileContents(file).then(function (hash) {
+    var parsedPath = path.parse(file);
+    return s3.upload({
+      Bucket: bucketName,
+      Key: path.format({
+        name: parsedPath.name + '-' + hash,
+        ext: parsedPath.ext
+      }),
+      Body: fs.createReadStream(file)
+    }).promise().then(function (result) {
+      progress(1, 1);
+      return result;
+    });
+  });
+}
+
+function waitForChangeSetComplete (cloudformation, stackName, changeSetName) {
+  return new Promise(function (resolve, reject) {
```

Comment:
Depending on the resources that you create or update this can sometimes take 5 minutes or even longer - I don't know what a reasonable timeout would be, therefore I propose to leave it as it is.